apis/v1/shared_types.go

```diff
@@ -745,7 +745,7 @@ const (
 )
 
 // SessionPersistence defines the desired state of SessionPersistence.
-// +kubebuilder:validation:XValidation:message="AbsoluteTimeout must be specified when cookie lifetimeType is Permanent",rule="!has(self.cookieConfig.lifetimeType) || self.cookieConfig.lifetimeType != 'Permanent' || has(self.absoluteTimeout)"
+// +kubebuilder:validation:XValidation:message="AbsoluteTimeout must be specified when cookie lifetimeType is Permanent",rule="self.type == 'Header' || !has(self.cookieConfig.lifetimeType) || self.cookieConfig.lifetimeType != 'Permanent' || has(self.absoluteTimeout)"
```

Comment:
The issue with the CEL config is that it assumes `self.cookieConfig` will always be set (which is not the case which is why it fail).

I believe we should not be special casing this for the `Header` type, but rather fix the underlying cause and use

```
!has(self.cookieConfig)
```

The eventual expression becomes:

```
!has(self.cookieConfig) || self.cookieConfig.lifetimeType != 'Permanent' || has(self.absoluteTimeout)
```